doc/source/merging.rst

```diff
@@ -301,6 +301,28 @@ This is also a valid argument to :meth:`DataFrame.append`:
           labels=['df1', 'df4'], vertical=True);
    plt.close('all');
 
+.. _merging.assign:
+
+Concatenating using ``assign``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can concatenate one Series to a DataFrame using the DataFrame method
```

Comment:
I'm a bit worried about this phrasing, and the section title.

Concatenation already has a well-defined meaning in pandas. This is similar, since `df[col] = series`, which is what `assign` is doing, will do alignment like `concat`, but it's different since you're only adding a single column.

I'd almost prefer a simple like this at the start or end of the `concat` documentation that says something like "concat combins many pandas objects into a single new one. If you want to to include a Series in a DataFrame as a new column, assign directly like ..."

Reply:
Please read my [initial explanation](https://github.com/pandas-dev/pandas/pull/18665/files/742f4962a4e6f70621ff6b1949a3a86144967866#r155961428) as to why I believed this can send newcomers into the right direction quicker. My contribution was for that goal.

Now, it's been a while and I'm no longer new to this problem so I can't put myself back into that mindset anymore. All I remember is that the warning wouldn't go away and that the first solution I stumbled upon seemed too complex and `assign` should have been proposed, while reading that merging documentation.

> If you want to to include a Series in a DataFrame as a new column, assign directly like ...

Is exactly what the [section right after my contribution](https://pandas.pydata.org/pandas-docs/stable/merging.html#concatenating-with-mixed-ndims) proposes yet it is still in the `merging` doc. I've put a reference to `assign` in front of it to redirect people to a cleaner solution when they have that specific problem.

Moving towards patch acceptance: if I change `Concatenating using ` to `Add a Series to a DataFrame using ` in the title and in the text body, would that satisfy your concern?

Thanks for your feedback!

Reply:
rather than making a separate section, I would be ok with making this a ::note.. I think. Its just a terminology issue, might be helpful to understand what concatenate is different from assign.